shared/configHelper.js

```diff
@@ -44,7 +44,7 @@ function readConfig(args, toolName, toolVersion, appTypes, handler) {
         process.exit(0);
         break;
     case 'create': 
-        processorList = createArgsProcessorList(appTypes); 
+        processorList = createArgsProcessorList(appTypes, false, toolsChecker); 
```

Comment:
When creating with a template, we won't know the app type until we download the template (which could be time consuming) - that's still happening once all the command line stuff is completed. And as a result, tools check in that case still happens after all the command line stuff is completed.